first-party/jib-layer-filter-extension-maven/src/main/java/com/google/cloud/tools/jib/maven/extension/layerfilter/JibLayerFilterExtension.java

```diff
@@ -102,9 +122,120 @@ public ContainerBuildPlan extendContainerBuildPlan(
         .map(FileEntriesLayer.Builder::build)
         .filter(layer -> !layer.getEntries().isEmpty())
         .forEach(newPlanBuilder::addLayer);
+
+    ContainerBuildPlan newPlan = newPlanBuilder.build();
+
+    if (config.get().isCreateParentLayers()) {
+      newPlan = moveParentDepsToNewLayers(newPlan, mavenData, logger);
+    }
+
+    return newPlan;
+  }
+
+  private ContainerBuildPlan moveParentDepsToNewLayers(
+      ContainerBuildPlan buildPlan, MavenData mavenData, ExtensionLogger logger)
+      throws JibPluginExtensionException {
+
+    logger.log(LogLevel.LIFECYCLE, "Moving parent dependencies to new layers.");
+
+    // the key is the expected path for the parent dependency
+    Map<String, Artifact> parentDependencies =
+        getParentDependencies(mavenData)
+            .stream()
+            .map(d -> d.getArtifact())
+            // TODO: configurable?
+            .collect(
+                // TODO: getVersion or getBaseVersion? check in jib, which version it is...
+                Collectors.toMap(
+                    a -> "/app/libs/" + a.getArtifactId() + "-" + a.getVersion() + ".jar", a -> a));
```

Comment:
`artifact` instead of `a`.

Note that the the app root (`/app`) is configurable in Jib, so ideally, we should check the config value as done in other plugins [like this](https://github.com/GoogleContainerTools/jib-extensions/blob/1a79b4d5121be5f09679f1f387f8b2edecf556b2/first-party/jib-quarkus-extension-maven/src/main/java/com/google/cloud/tools/jib/maven/extension/quarkus/JibQuarkusExtension.java#L174-L175).

If you don't want to address this here, that's fine. Just file a new issue to track this limitation.

Reply:
I have no better idea yet but still thinking about it.

Reply:
Okay, with my latest commit https://github.com/GoogleContainerTools/jib-extensions/pull/76/commits/469cd366d7ab6723bb8e059ff65ef4925775a21c , I only consider filenames when searching for a match. I am fine with this approach if you are.

Reply:
With https://github.com/GoogleContainerTools/jib-extensions/pull/76/commits/1a5d3210c2aaedad2e96401d48d1c22cfc7404cd I now use source file path instead of getting the filename from extraction path. I think this is the correct way, especially since jib applies some de-duplication to the extraction path here:
https://github.com/GoogleContainerTools/jib/blob/v2.8.0-maven/jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java#L598

Reply:
With https://github.com/GoogleContainerTools/jib-extensions/commit/d974582e3e6fed55c0235e7cafca604945184765 the full source path is matched now. This is always unique and especially contains the group id.